crates/rattler_installs_packages/src/venv.rs

```diff
@@ -0,0 +1,177 @@
+//! Module that helps with allowing in the creation of python virtual environments.
+//! Now just use the python venv command to create the virtual environment.
+//! Later on we can look into actually creating the environment by linking to the python library,
+//! and creating the necessary files. See: [VEnv](https://packaging.python.org/en/latest/specifications/virtual-environments/#declaring-installation-environments-as-python-virtual-environments)
+#![allow(dead_code)]
+use crate::system_python::{
+    system_python_executable, FindPythonError, ParsePythonInterpreterVersionError,
+    PythonInterpreterVersion,
+};
+use crate::wheel::UnpackError;
+use crate::{InstallPaths, UnpackWheelOptions, Wheel};
+use std::path::{Path, PathBuf};
+use std::process::{Command, Output};
+use thiserror::Error;
+
+/// Specifies where to find the python executable
+pub enum PythonLocation {
+    /// Use system interpreter
+    System,
+    // Use custom interpreter
+    Custom(PathBuf),
+}
+
+impl PythonLocation {
+    /// Location of python executable
+    pub fn executable(&self) -> Result<PathBuf, FindPythonError> {
+        match self {
+            PythonLocation::System => system_python_executable(),
+            PythonLocation::Custom(path) => Ok(path.clone()),
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum VEnvError {
+    #[error(transparent)]
+    FindPythonError(#[from] FindPythonError),
+    #[error(transparent)]
+    ParsePythonInterpreterVersionError(#[from] ParsePythonInterpreterVersionError),
+    #[error("failed to run 'python -m venv': `{0}`")]
+    FailedToRun(String),
+    #[error(transparent)]
+    FailedToCreate(#[from] std::io::Error),
+}
+
+/// Represents a virtual environment in which wheels can be installed
+pub struct VEnv {
+    /// Location of the virtual environment
+    location: PathBuf,
+    /// Install paths for this virtual environment
+    install_paths: InstallPaths,
+    /// Specifies if this is a windows venv
+    windows: bool,
```

Comment:
Should we maybe store this information in the `InstallPaths` struct?

Reply:
Refactored install paths! Take a look!